### PR TITLE
Enhancement/react18

### PR DESCRIPTION
**Pull request by jordanlesich**
## GitHub Issue

https://github.com/HausDAO/daohaus-monorepo/issues/564

## Changes

- Update React App Main.tsx to use react-dom client instead of react-dom
- Needed new react-dom types for some reason

- Updated some useEffects, though it was 

## Packages Added

Brief list of any packages added, and if folks need to rerun `yarn install` to run locally

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: apps/core-app/src/main.tsx [hunk 1]
```diff
@@ -1,19 +1,27 @@
+import { StrictMode } from 'react';
+import { createRoot } from 'react-dom/client';
```
Comment (0xRowdy) [inline apps/core-app/src/main.tsx:2]: This applies to below comments to remove disable of eslint.
```suggestion
import ReactDOM from 'react-dom/client';
```

File: apps/core-app/src/main.tsx [hunk 2]
```diff
@@ -1,19 +1,27 @@
...
+
 import Routes from './Routes';
 
-ReactDOM.render(
+const container = document.getElementById('root');
+
+//reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis
+// THis is how react wants to render the app.
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const root = createRoot(container!);
```
Comment (0xRowdy) [inline apps/core-app/src/main.tsx:15]:
```suggestion
const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
```

File: apps/hub-app/src/main.tsx [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 
 import Routes from './Routes';
 
-ReactDOM.render(
+const container = document.getElementById('root');
+
+//reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis
+// THis is how react wants to render the app.
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const root = createRoot(container!);
```
Comment (0xRowdy) [inline apps/hub-app/src/main.tsx:15]: Sames as in core

File: apps/summon-app/src/main.tsx [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
 
 import App from './app/App';
 
-ReactDOM.render(
+const container = document.getElementById('root');
+
+//reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis
+// THis is how react wants to render the app.
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const root = createRoot(container!);
```
Comment (0xRowdy) [inline apps/summon-app/src/main.tsx:15]: Same

**Review by 0xRowdy — COMMENTED** [review]
I tryed it out and didn't have any issues. LMK what you think or if there are problems I didn't see.

File: apps/summon-app/src/main.tsx (hunk 4 above)
Comment (jordanlesich) [reply apps/summon-app/src/main.tsx:15]: Yeah, I didn't actually get any problems when implementing it. Was just following the docs as they had recommended it if we're using TS, and our linter doesn't allow type assertions. 

Figured I would follow the docs on the small chance that there's some sort of downstream linting headache that emerges out of the blue.

Comment (0xRowdy) [reply apps/summon-app/src/main.tsx:15]: Yeah, I noticed that. 
If you do Create React App with Typescript it does it the way I posted. Odd that it isn't consistent with the docs.

**Review by 0xRowdy — APPROVED** [review]

**Pull request closed without merging**